src/server/gc_worker/applied_lock_collector.rs

```diff
@@ -25,14 +25,17 @@ const MAX_COLLECT_SIZE: usize = 1024;
 #[derive(Default)]
 struct LockObserverState {
     max_ts: AtomicU64,
+
+    /// `is_clean` is true, only it's sure that all applying of stale locks (locks with start_ts <=
+    /// specified max_ts) are monitored and collected. If there are too many stale locks or any
+    /// error happens, `is_clean` must be set to `false`.
+    is_clean: AtomicBool,
 }
 
 #[derive(Debug)]
 enum LockObserverMsg {
```

Comment:
What other messages can there be in the future? If there isn't any, can we just get rid of a new enum or struct?

Reply:
Ah, yes